### PR TITLE
move gear, batch_client, pipeline to hailtop

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

**Comment by cseed** [comment]
@johnc1231 You should be blocked on `hailctl batch` once this goes in.

File: hail/python/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+aiohttp
```
Comment (cseed) [inline hail/python/requirements.txt]: From https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires:

"It is not considered best practice to use install_requires to pin dependencies to specific versions, or to specify sub-dependencies (i.e. dependencies of your dependencies). This is overly-restrictive, and prevents the user from gaining the benefit of dependency upgrades."

I don't think we should be (semi) pinning dependencies here unless (1) we require a minimum version, or (2) there are specific versions we know don't work.  The one exception I'm aware of is pyspark, which we need to specify precisely for jar compatibility.

@tpoterba does that sound right?  In which case, I'm going to unpin these packages.

**Comment by cseed** [comment]
Also moved pipeline to hailtop so it is including in the Hail package.  The tests remain in the pipeline/ subproject directory.  @konradjk You'll have to import pipeline with `from hailtop import pipeline` after this.

FYI @jigold

File: hail/python/requirements.txt [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-pyspark>=2.4,<2.4.2
-parsimonious<0.9
-ipykernel<5
 decorator<5
-requests>=2.21.0,<2.21.1
 gcsfs==0.2.1
 hurry.filesize==0.9
-scipy>1.2,<1.4
\ No newline at end of file
+ipykernel<5
```
Comment (johnc1231) [inline hail/python/requirements.txt]: I believe you're missing jwt here. If I check out this branch, `make-install-wheel`, and then do from `hailtop import pipeline` I get a `ModuleNotFoundError: No module named 'jwt'`.

**Pull request merged**